packages/color-scale-block/src/components/AddNewColor.tsx

```diff
@@ -0,0 +1,87 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { FC, useState } from "react";
+import { AddNewColorModalProps, ColorProps } from "../types";
+import {
+    Flyout,
+    FlyoutPlacement,
+    ColorPicker,
+    ColorFormat,
+    Color,
+} from "@frontify/fondue";
+import { uuid } from "uuidv4";
+
+export const AddNewColor: FC<AddNewColorModalProps> = ({
+    newIndex,
+    colorPickerRef,
+    updateColor,
+    editedColor,
+    setEditedColor,
+    isColorPickerOpen,
+    setIsColorPickerOpen,
+    setFormat,
+    colors,
+}) => {
+    const [colorPickerFormat, setColorPickerFormat] = useState(ColorFormat.Hex);
+
+    return (
+        <>
+            <div ref={colorPickerRef}>
+                <Flyout
+                    placement={FlyoutPlacement.Top}
+                    isOpen={isColorPickerOpen}
+                    onCancel={() => {
+                        setIsColorPickerOpen(false);
```

Comment:
For **all** handler functions in **all files**, create proper handlers outside of JSX. Except for onOpenChange, name it noop or leave it as it is if you use it only once.
Example:
```
handleFlyoutCancel = () => {
  setIsColorPickerOpen(false);
  setEditedColor(null);
}
```
And then use it as
`onCancel={handleFlyoutCancel}`